presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -856,10 +856,18 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalBinaryExpression(LogicalBinaryExpression node, Object context)
         {
+            LogicalBinaryExpression.Type type = node.getType();
```

Comment:
cmt msg
> Optimize ExpressionInterpreter for predicate expressions

change to sth like `Short-circuit logical expression in ExpressionInterpreter`